[V2] Update .gitignore to include all automatically-generated files/folders

skmendez opened the pull request:
### Type

- [X] Bugfix
- [X] Enhancement
- [ ] New feature

### Description of the changes
Whenever Red is ran, the two files `start_red.bat` and `start_red_autorestart.bat` will be automatically generated, and `lib` will be populated when requirements are installed. Also in the case of OSX/Unix, the files `start_launcher.command` and `start_launcher.sh` will be generated, respectively. This patch will cause `.gitignore` to not include these files/folders automatically on commit.

skmendez committed:
Add start_red*.bat to .gitignore

Whenever Red is ran, the two files `start_red.bat` and `start_red_autorestart.bat` will be automatically generated. This will cause `.gitignore` to not include these files automatically on commit.

Kowlin commented:
I honestly don't see a reason for adding this to the ``.gitignore``. Red isn't meant to be re-uploaded to a git source.

skmendez committed:
Adds lib, and non-Windows start-launchers

Any file automatically generated by running Red should now be taken care of by the `.gitignore`.
